crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1032,6 +1032,368 @@ impl<'w> EntityWorldMut<'w> {
     pub fn update_location(&mut self) {
         self.location = self.world.entities().get(self.entity).unwrap();
     }
+
+    /// Gets an Entry into the world for this entity and component for in-place manipulation.
```

Comment:
Would be nice to have some example here as I bet it's the entry point from which you start exploring this API:

```suggestion
    /// Gets an Entry into the world for this entity and component for in-place manipulation.
    ///
    /// The type parameter specifies which component to get.
    ///
    /// # Examples
    ///
    /// ```
    /// # use bevy_ecs::prelude::*;
    /// #[derive(Component, Default, Clone, Copy, Debug, PartialEq)]
    /// struct Comp(u32);
    ///
    /// let mut world = World::new();
    ///
    /// let mut entity = world.spawn(Comp(0));
    /// entity.entry::<Comp>().and_modify(|mut c| c.0 += 1);
    /// # assert_eq!(world.query::<&Comp>().single(&world).0, 1);
    ///
    /// let mut new_entity = world.spawn_empty();
    /// entity.entry().or_insert_with(|| Comp(4));
    /// # assert_eq!(world.query::<&Comp>().single(&world).0, 4);
    /// ```
```